packages/tools-iottwinmaker/src/lib/workspace.ts

```diff
@@ -0,0 +1,182 @@
+    RoleName: roleName,
+    Path: '/',
+    AssumeRolePolicyDocument: roleAssumePolicy,
+    Description: `${roleType} generated by AWS IoT TwinMaker Getting Started Script`,
+  });
+
+  if (!createRoleResponse.Role || !createRoleResponse.Role.Arn) {
+    throw new Error(`Failed to create role ${roleName}.`);
+  }
+
+  const roleArn: string = createRoleResponse.Role.Arn;
+  console.log(`Created Role: ${roleArn}.`);
+  const createPolicyResponse: CreatePolicyCommandOutput = await aws().iam.createPolicy({
+    PolicyName: `IoTTwinMaker${roleType}Policy-${uniqueId}`,
+    PolicyDocument: rolePolicy,
+    Description: `${roleType} Policy generated by AWS IoT TwinMaker Getting Started Script`,
+  });
+
+  const policyArn: string | undefined = createPolicyResponse.Policy?.Arn;
+  console.log(`Created Policy: ${policyArn}`);
+  await aws().iam.attachRolePolicy({
+    RoleName: roleName,
+    PolicyArn: policyArn,
+  });
+  console.log(`Attached Policy: ${policyArn} to Role: ${roleArn}.`);
+  return roleArn;
+}
+
+/**
+ * Helper function during workspace creation to create S3 bucket
+ * @param uniqueId identifier for bucket used as suffix in the name
+ * @returns promise of the role S3 bucket name and arn
+ */
+async function createWorkspaceS3Bucket(uniqueId: string) {
+  const s3BucketName = `iot-twinmaker-workspace-bucket-${uniqueId}`;
+  console.log('Creating S3 Bucket for the TwinMaker Workspace...');
+  const createBucketResponse: CreateBucketCommandOutput = await aws().s3.createBucket({ Bucket: s3BucketName });
+  console.log(`Workspace S3 Bucket created: ${createBucketResponse.Location}`);
+  console.log('Configuring CORS policy...');
+  aws().s3.putBucketCors({
+    Bucket: s3BucketName,
+    CORSConfiguration: {
+      CORSRules: [
+        {
+          AllowedHeaders: ['*'],
+          AllowedMethods: ['GET', 'PUT', 'POST', 'DELETE', 'HEAD'],
+          AllowedOrigins: ['*'],
+          ExposeHeaders: ['ETag', 'Access-Control-Allow-Origin'],
+        },
+      ],
+    },
+  });
+  return { s3BucketName, s3BucketArn: `arn:aws:s3:::${s3BucketName}` };
+}
+
+/**
+ * Driver function for workspace creation
+ * @param workspaceId workspace-id to be created
+ * @returns promise of workspace arn, s3 arn, role arn, dashboardrole arn
+ */
+async function prepareWorkspace(workspaceId: string) {
+  const uniqueId: string = createUniqueId();
+  const region: string = aws().region;
+  const { accountId, accountArn } = await aws().getCurrentIdentity();
+  const { s3BucketArn } = await createWorkspaceS3Bucket(uniqueId);
+
+  const policyParams: Record<string, string> = {
+    accountId,
+    accountArn,
+    region,
+    workspaceS3BucketArn: s3BucketArn,
+  };
+
+  const workspaceRolePolicy: string = replaceTemplateVars(JSON.stringify(workspaceRolePolicyTemplate), policyParams);
+  const workspaceRoleArn: string = await createRoleAndPolicy(
+    uniqueId,
+    'WorkspaceRole',
```

Comment:
Need a more specific name for the policy than this, maybe we can follow the standard twinmaker pattern of:
`twinmaker-workspace-<workspaceId>-<accountId>-<region>`

Reply:
+1, similar to other comment, I think ideally the TMDT create workspace should result in the same as if it was done using the console

Reply:
makes sense, updated